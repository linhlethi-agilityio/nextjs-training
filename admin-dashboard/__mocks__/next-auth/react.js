export const useSession = () => ({
  data: {
    user: {
      token: 'access-token',
      email: 'user@gmail.com',
      id: '1',
      name: 'User',
      role: 'admin',
    },
    expires: Date.now().toString(),
  },
  status: 'authenticated',
  update: async () => null,
});
