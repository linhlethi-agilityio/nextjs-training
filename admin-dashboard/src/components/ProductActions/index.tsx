'use client';

import { memo } from 'react';
import { Button, Flex } from '@chakra-ui/react';

// Icons
import { FileArrowUpIcon, PrinterIcon, CreateFolderIcon } from '@/icons';

// Components
import { SearchInput } from '@/components';

const ProductActions = () => {
  const handleChangeSearchInput = () => {
    //TODO: update later
  };

  return (
    <Flex justifyContent="space-between" mt={26} ml={8}>
      <SearchInput onChange={handleChangeSearchInput} w={465} />
      <Flex gap={25}>
        <Button variant="outline" rightIcon={<FileArrowUpIcon />}>
          Export
        </Button>
        <Button variant="outline" rightIcon={<PrinterIcon />}>
          Print
        </Button>
        <Button rightIcon={<CreateFolderIcon />} colorScheme="brand" px={25}>
          Create Folder
        </Button>
      </Flex>
    </Flex>
  );
};

export default memo(ProductActions);
