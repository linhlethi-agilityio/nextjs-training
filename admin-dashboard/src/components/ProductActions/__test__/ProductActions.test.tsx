import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import ProductActions from '..';

const mockReplace = jest.fn();
const mockPathname = 'localhost:3000';
const mockURLSearchParams = {
  page: '1',
};

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  usePathname: jest.fn(() => mockPathname),
  useRouter: jest.fn(() => ({ replace: mockReplace })),
  useSearchParams: jest.fn(() => new URLSearchParams(mockURLSearchParams)),
}));

const mockAddOrderAction = jest.fn();

const mockProps = {
  addOrderAction: mockAddOrderAction,
};

describe('ProductActions component', () => {
  it('should match snapshot for ProductActions', () => {
    const { container } = render(<ProductActions {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
