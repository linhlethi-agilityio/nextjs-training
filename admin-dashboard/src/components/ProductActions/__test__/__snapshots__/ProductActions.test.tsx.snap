// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductActions component should match snapshot for ProductActions 1`] = `
<div>
  <div
    class="css-aniilw"
  >
    <div
      class="chakra-input__group css-ck76jc"
      data-group="true"
    >
      <div
        class="chakra-input__left-element css-2lnh3j"
        style="cursor: pointer;"
      >
        <svg
          class="chakra-icon css-nyt2fo"
          focusable="false"
          viewBox="0 0 17 16"
        >
          <g
            clip-path="url(#clip0_16_102)"
          >
            <path
              d="M12.0725 10.344C13.0408 9.0227 13.4745 7.3845 13.2869 5.75716C13.0992 4.12982 12.304 2.63335 11.0603 1.56714C9.81666 0.50093 8.2163 -0.0563875 6.57938 0.00668566C4.94247 0.0697588 3.38973 0.748571 2.23182 1.90732C1.0739 3.06606 0.396199 4.61929 0.334297 6.25624C0.272396 7.8932 0.830858 9.49317 1.89796 10.7361C2.96505 11.9789 4.46209 12.7731 6.08957 12.9596C7.71704 13.1461 9.35493 12.7112 10.6755 11.742H10.6745C10.7045 11.782 10.7365 11.82 10.7725 11.857L14.6225 15.707C14.8101 15.8946 15.0644 16.0001 15.3297 16.0002C15.595 16.0003 15.8494 15.895 16.037 15.7075C16.2247 15.52 16.3302 15.2656 16.3303 15.0004C16.3303 14.7351 16.2251 14.4806 16.0375 14.293L12.1875 10.443C12.1518 10.4068 12.1134 10.3734 12.0725 10.343V10.344ZM12.3305 6.50001C12.3305 7.22228 12.1883 7.93748 11.9119 8.60477C11.6355 9.27206 11.2304 9.87837 10.7196 10.3891C10.2089 10.8998 9.6026 11.3049 8.93531 11.5813C8.26802 11.8577 7.55282 12 6.83055 12C6.10828 12 5.39308 11.8577 4.72579 11.5813C4.0585 11.3049 3.45218 10.8998 2.94146 10.3891C2.43074 9.87837 2.02561 9.27206 1.74921 8.60477C1.47281 7.93748 1.33055 7.22228 1.33055 6.50001C1.33055 5.04132 1.91001 3.64237 2.94146 2.61092C3.97291 1.57947 5.37186 1.00001 6.83055 1.00001C8.28924 1.00001 9.68819 1.57947 10.7196 2.61092C11.7511 3.64237 12.3305 5.04132 12.3305 6.50001Z"
              fill="currentColor"
            />
          </g>
          <defs>
            <clippath
              id="clip0_16_102"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0.330566)"
                width="16"
              />
            </clippath>
          </defs>
        </svg>
      </div>
      <input
        class="chakra-input css-1ecsgz3"
        placeholder="Search by ID Order"
        value=""
      />
      <div
        class="chakra-input__right-element css-czbnbi"
        style="cursor: pointer;"
      >
        <svg
          class="chakra-icon css-nyt2fo"
          focusable="false"
          viewBox="0 0 17 16"
        >
          <g
            clip-path="url(#clip0_16_105)"
          >
            <path
              clip-rule="evenodd"
              d="M11.8306 2C11.6336 2 11.4385 2.0388 11.2565 2.11418C11.0746 2.18957 10.9092 2.30006 10.7699 2.43934C10.6306 2.57863 10.5201 2.74399 10.4447 2.92598C10.3694 3.10797 10.3306 3.30302 10.3306 3.5C10.3306 3.69699 10.3694 3.89204 10.4447 4.07403C10.5201 4.25602 10.6306 4.42138 10.7699 4.56066C10.9092 4.69995 11.0746 4.81044 11.2565 4.88582C11.4385 4.96121 11.6336 5 11.8306 5C12.2284 5 12.6099 4.84197 12.8912 4.56066C13.1725 4.27936 13.3306 3.89783 13.3306 3.5C13.3306 3.10218 13.1725 2.72065 12.8912 2.43934C12.6099 2.15804 12.2284 2 11.8306 2ZM9.38057 3C9.49532 2.43486 9.80193 1.92677 10.2484 1.56181C10.6949 1.19686 11.2539 0.997498 11.8306 0.997498C12.4072 0.997498 12.9662 1.19686 13.4127 1.56181C13.8592 1.92677 14.1658 2.43486 14.2806 3H16.3306V4H14.2806C14.1658 4.56515 13.8592 5.07324 13.4127 5.43819C12.9662 5.80314 12.4072 6.00251 11.8306 6.00251C11.2539 6.00251 10.6949 5.80314 10.2484 5.43819C9.80193 5.07324 9.49532 4.56515 9.38057 4H0.330566V3H9.38057ZM4.83057 7C4.43274 7 4.05121 7.15804 3.76991 7.43934C3.4886 7.72065 3.33057 8.10218 3.33057 8.5C3.33057 8.89783 3.4886 9.27936 3.76991 9.56066C4.05121 9.84197 4.43274 10 4.83057 10C5.22839 10 5.60992 9.84197 5.89123 9.56066C6.17253 9.27936 6.33057 8.89783 6.33057 8.5C6.33057 8.10218 6.17253 7.72065 5.89123 7.43934C5.60992 7.15804 5.22839 7 4.83057 7ZM2.38057 8C2.49532 7.43486 2.80193 6.92677 3.24844 6.56181C3.69494 6.19686 4.25389 5.9975 4.83057 5.9975C5.40724 5.9975 5.96619 6.19686 6.4127 6.56181C6.8592 6.92677 7.16581 7.43486 7.28057 8H16.3306V9H7.28057C7.16581 9.56515 6.8592 10.0732 6.4127 10.4382C5.96619 10.8031 5.40724 11.0025 4.83057 11.0025C4.25389 11.0025 3.69494 10.8031 3.24844 10.4382C2.80193 10.0732 2.49532 9.56515 2.38057 9H0.330566V8H2.38057ZM11.8306 12C11.4327 12 11.0512 12.158 10.7699 12.4393C10.4886 12.7206 10.3306 13.1022 10.3306 13.5C10.3306 13.8978 10.4886 14.2794 10.7699 14.5607C11.0512 14.842 11.4327 15 11.8306 15C12.2284 15 12.6099 14.842 12.8912 14.5607C13.1725 14.2794 13.3306 13.8978 13.3306 13.5C13.3306 13.1022 13.1725 12.7206 12.8912 12.4393C12.6099 12.158 12.2284 12 11.8306 12ZM9.38057 13C9.49532 12.4349 9.80193 11.9268 10.2484 11.5618C10.6949 11.1969 11.2539 10.9975 11.8306 10.9975C12.4072 10.9975 12.9662 11.1969 13.4127 11.5618C13.8592 11.9268 14.1658 12.4349 14.2806 13H16.3306V14H14.2806C14.1658 14.5651 13.8592 15.0732 13.4127 15.4382C12.9662 15.8031 12.4072 16.0025 11.8306 16.0025C11.2539 16.0025 10.6949 15.8031 10.2484 15.4382C9.80193 15.0732 9.49532 14.5651 9.38057 14H0.330566V13H9.38057Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </g>
          <defs>
            <clippath
              id="clip0_16_105"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0.330566)"
                width="16"
              />
            </clippath>
          </defs>
        </svg>
      </div>
    </div>
    <div
      class="css-1sb31b2"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Export
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-nyt2fo"
            focusable="false"
            viewBox="0 0 13 16"
          >
            <g
              clip-path="url(#clip0_16_125)"
            >
              <path
                d="M8.91637 11C9.04898 11 9.17616 10.9473 9.26993 10.8536C9.36369 10.7598 9.41637 10.6326 9.41637 10.5V6.70701L10.5624 7.85401C10.6563 7.9479 10.7836 8.00064 10.9164 8.00064C11.0491 8.00064 11.1765 7.9479 11.2704 7.85401C11.3643 7.76012 11.417 7.63278 11.417 7.50001C11.417 7.36723 11.3643 7.2399 11.2704 7.14601L9.27037 5.14601C9.22393 5.09945 9.16875 5.0625 9.10801 5.0373C9.04726 5.01209 8.98214 4.99911 8.91637 4.99911C8.85061 4.99911 8.78548 5.01209 8.72474 5.0373C8.66399 5.0625 8.60882 5.09945 8.56237 5.14601L6.56237 7.14601C6.46849 7.2399 6.41574 7.36723 6.41574 7.50001C6.41574 7.63278 6.46849 7.76012 6.56237 7.85401C6.65626 7.9479 6.7836 8.00064 6.91637 8.00064C7.04915 8.00064 7.17649 7.9479 7.27037 7.85401L8.41637 6.70701V10.5C8.41637 10.6326 8.46905 10.7598 8.56282 10.8536C8.65659 10.9473 8.78376 11 8.91637 11Z"
                fill="currentColor"
              />
              <path
                d="M4.91638 0C4.38595 0 3.87724 0.210714 3.50217 0.585786C3.1271 0.960859 2.91638 1.46957 2.91638 2V14C2.91638 14.5304 3.1271 15.0391 3.50217 15.4142C3.87724 15.7893 4.38595 16 4.91638 16H12.9164C13.4468 16 13.9555 15.7893 14.3306 15.4142C14.7057 15.0391 14.9164 14.5304 14.9164 14V2C14.9164 1.46957 14.7057 0.960859 14.3306 0.585786C13.9555 0.210714 13.4468 0 12.9164 0L4.91638 0ZM4.91638 1H12.9164C13.1816 1 13.436 1.10536 13.6235 1.29289C13.811 1.48043 13.9164 1.73478 13.9164 2V14C13.9164 14.2652 13.811 14.5196 13.6235 14.7071C13.436 14.8946 13.1816 15 12.9164 15H4.91638C4.65117 15 4.39681 14.8946 4.20928 14.7071C4.02174 14.5196 3.91638 14.2652 3.91638 14V2C3.91638 1.73478 4.02174 1.48043 4.20928 1.29289C4.39681 1.10536 4.65117 1 4.91638 1Z"
                fill="currentColor"
              />
            </g>
            <defs>
              <clippath
                id="clip0_16_125"
              >
                <rect
                  fill="white"
                  height="16"
                  transform="translate(0.916382)"
                  width="16"
                />
              </clippath>
            </defs>
          </svg>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Print
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-nyt2fo"
            focusable="false"
            viewBox="0 0 17 17"
          >
            <g
              clip-path="url(#clip0_16_130)"
            >
              <path
                d="M3.41638 8.5C3.54899 8.5 3.67617 8.44732 3.76994 8.35355C3.8637 8.25979 3.91638 8.13261 3.91638 8C3.91638 7.86739 3.8637 7.74021 3.76994 7.64645C3.67617 7.55268 3.54899 7.5 3.41638 7.5C3.28377 7.5 3.1566 7.55268 3.06283 7.64645C2.96906 7.74021 2.91638 7.86739 2.91638 8C2.91638 8.13261 2.96906 8.25979 3.06283 8.35355C3.1566 8.44732 3.28377 8.5 3.41638 8.5Z"
                fill="currentColor"
              />
              <path
                d="M5.91638 1.5C5.38595 1.5 4.87724 1.71071 4.50217 2.08579C4.1271 2.46086 3.91638 2.96957 3.91638 3.5V5.5H2.91638C2.38595 5.5 1.87724 5.71071 1.50217 6.08579C1.1271 6.46086 0.916382 6.96957 0.916382 7.5L0.916382 10.5C0.916382 11.0304 1.1271 11.5391 1.50217 11.9142C1.87724 12.2893 2.38595 12.5 2.91638 12.5H3.91638V13.5C3.91638 14.0304 4.1271 14.5391 4.50217 14.9142C4.87724 15.2893 5.38595 15.5 5.91638 15.5H11.9164C12.4468 15.5 12.9555 15.2893 13.3306 14.9142C13.7057 14.5391 13.9164 14.0304 13.9164 13.5V12.5H14.9164C15.4468 12.5 15.9555 12.2893 16.3306 11.9142C16.7057 11.5391 16.9164 11.0304 16.9164 10.5V7.5C16.9164 6.96957 16.7057 6.46086 16.3306 6.08579C15.9555 5.71071 15.4468 5.5 14.9164 5.5H13.9164V3.5C13.9164 2.96957 13.7057 2.46086 13.3306 2.08579C12.9555 1.71071 12.4468 1.5 11.9164 1.5H5.91638ZM4.91638 3.5C4.91638 3.23478 5.02174 2.98043 5.20928 2.79289C5.39681 2.60536 5.65117 2.5 5.91638 2.5H11.9164C12.1816 2.5 12.436 2.60536 12.6235 2.79289C12.811 2.98043 12.9164 3.23478 12.9164 3.5V5.5H4.91638V3.5ZM5.91638 8.5C5.38595 8.5 4.87724 8.71071 4.50217 9.08579C4.1271 9.46086 3.91638 9.96957 3.91638 10.5V11.5H2.91638C2.65117 11.5 2.39681 11.3946 2.20928 11.2071C2.02174 11.0196 1.91638 10.7652 1.91638 10.5V7.5C1.91638 7.23478 2.02174 6.98043 2.20928 6.79289C2.39681 6.60536 2.65117 6.5 2.91638 6.5H14.9164C15.1816 6.5 15.436 6.60536 15.6235 6.79289C15.811 6.98043 15.9164 7.23478 15.9164 7.5V10.5C15.9164 10.7652 15.811 11.0196 15.6235 11.2071C15.436 11.3946 15.1816 11.5 14.9164 11.5H13.9164V10.5C13.9164 9.96957 13.7057 9.46086 13.3306 9.08579C12.9555 8.71071 12.4468 8.5 11.9164 8.5H5.91638ZM12.9164 10.5V13.5C12.9164 13.7652 12.811 14.0196 12.6235 14.2071C12.436 14.3946 12.1816 14.5 11.9164 14.5H5.91638C5.65117 14.5 5.39681 14.3946 5.20928 14.2071C5.02174 14.0196 4.91638 13.7652 4.91638 13.5V10.5C4.91638 10.2348 5.02174 9.98043 5.20928 9.79289C5.39681 9.60536 5.65117 9.5 5.91638 9.5H11.9164C12.1816 9.5 12.436 9.60536 12.6235 9.79289C12.811 9.98043 12.9164 10.2348 12.9164 10.5Z"
                fill="currentColor"
              />
            </g>
            <defs>
              <clippath
                id="clip0_16_130"
              >
                <rect
                  fill="white"
                  height="16"
                  transform="translate(0.916382 0.5)"
                  width="16"
                />
              </clippath>
            </defs>
          </svg>
        </span>
      </button>
      <button
        class="chakra-button css-a8a4fp"
        type="button"
      >
        Create Folder
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-nyt2fo"
            focusable="false"
            viewBox="0 0 17 16"
          >
            <g
              clip-path="url(#clip0_16_135)"
            >
              <path
                d="M3.41638 0C3.25038 0 3.08638 0.016 2.92938 0.048L3.12338 1.028C3.2199 1.00915 3.31804 0.99977 3.41638 1H3.87438V0H3.41638ZM5.70838 0H4.79138V1H5.70838V0ZM7.54138 0H6.62438V1H7.54138V0ZM9.37438 0H8.45838V1H9.37438V0ZM11.2084 0H10.2914V1H11.2084V0ZM13.0414 0H12.1244V1H13.0414V0ZM14.4164 0H13.9584V1H14.4164C14.5164 1 14.6154 1.01 14.7094 1.029L14.9034 0.048C14.743 0.0161758 14.5799 9.94507e-05 14.4164 0V0ZM16.4954 1.11C16.3126 0.837453 16.0782 0.603373 15.8054 0.421L15.2494 1.252C15.4134 1.362 15.5544 1.503 15.6644 1.667L16.4944 1.111L16.4954 1.11ZM2.02638 0.421C1.75383 0.603769 1.51975 0.838188 1.33738 1.111L2.16838 1.667C2.27838 1.503 2.41938 1.362 2.58338 1.252L2.02638 0.422V0.421ZM16.9164 2.5C16.9164 2.334 16.9004 2.17 16.8684 2.013L15.8884 2.207C15.9064 2.301 15.9164 2.399 15.9164 2.5V2.958H16.9164V2.5ZM0.964382 2.013C0.932558 2.17338 0.916481 2.33649 0.916382 2.5L0.916382 2.958H1.91638V2.5C1.91638 2.4 1.92638 2.301 1.94538 2.207L0.964382 2.013ZM0.916382 3.875V4.792H1.91638V3.875H0.916382ZM16.9164 4.792V3.875H15.9164V4.792H16.9164ZM0.916382 5.708V6.625H1.91638V5.708H0.916382ZM16.9164 6.625V5.708H15.9164V6.625H16.9164ZM0.916382 7.542V8.458H1.91638V7.542H0.916382ZM15.9164 8.458H16.9164V7.542H15.9164V8.458ZM0.916382 9.375V10.292H1.91638V9.375H0.916382ZM16.9164 10.292V9.375H15.9164V10.292H16.9164ZM0.916382 11.208V12.125H1.91638V11.208H0.916382ZM16.9164 12.125V11.208H15.9164V12.125H16.9164ZM0.916382 13.042V13.5C0.916382 13.666 0.932382 13.83 0.964382 13.987L1.94438 13.793C1.92553 13.6965 1.91615 13.5983 1.91638 13.5V13.042H0.916382ZM16.9164 13.5V13.042H15.9164V13.5C15.9164 13.6 15.9064 13.699 15.8874 13.793L16.8684 13.987C16.9004 13.829 16.9164 13.667 16.9164 13.5ZM1.33738 14.89C1.52038 15.162 1.75438 15.396 2.02738 15.579L2.58338 14.748C2.41924 14.6381 2.27824 14.4971 2.16838 14.333L1.33838 14.889L1.33738 14.89ZM15.8064 15.579C16.0784 15.396 16.3124 15.162 16.4954 14.889L15.6644 14.333C15.5544 14.497 15.4134 14.638 15.2494 14.748L15.8054 15.578L15.8064 15.579ZM2.92938 15.952C3.08738 15.984 3.24938 16 3.41638 16H3.87438V15H3.41638C3.31638 15 3.21738 14.99 3.12338 14.971L2.92938 15.952ZM14.4164 16C14.5824 16 14.7464 15.984 14.9034 15.952L14.7094 14.972C14.6129 14.9909 14.5147 15.0002 14.4164 15H13.9584V16H14.4164ZM4.79138 16H5.70838V15H4.79138V16ZM6.62438 16H7.54138V15H6.62438V16ZM8.45838 15V16H9.37438V15H8.45838ZM10.2914 16H11.2084V15H10.2914V16ZM12.1244 16H13.0414V15H12.1244V16ZM9.41638 4.5C9.41638 4.36739 9.3637 4.24021 9.26994 4.14645C9.17617 4.05268 9.04899 4 8.91638 4C8.78377 4 8.6566 4.05268 8.56283 4.14645C8.46906 4.24021 8.41638 4.36739 8.41638 4.5V7.5H5.41638C5.28377 7.5 5.1566 7.55268 5.06283 7.64645C4.96906 7.74021 4.91638 7.86739 4.91638 8C4.91638 8.13261 4.96906 8.25979 5.06283 8.35355C5.1566 8.44732 5.28377 8.5 5.41638 8.5H8.41638V11.5C8.41638 11.6326 8.46906 11.7598 8.56283 11.8536C8.6566 11.9473 8.78377 12 8.91638 12C9.04899 12 9.17617 11.9473 9.26994 11.8536C9.3637 11.7598 9.41638 11.6326 9.41638 11.5V8.5H12.4164C12.549 8.5 12.6762 8.44732 12.7699 8.35355C12.8637 8.25979 12.9164 8.13261 12.9164 8C12.9164 7.86739 12.8637 7.74021 12.7699 7.64645C12.6762 7.55268 12.549 7.5 12.4164 7.5H9.41638V4.5Z"
                fill="white"
              />
            </g>
            <defs>
              <clippath
                id="clip0_16_135"
              >
                <rect
                  fill="white"
                  height="16"
                  transform="translate(0.916382)"
                  width="16"
                />
              </clippath>
            </defs>
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
